src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -32,17 +32,19 @@ public static class ActivatorUtilities
             params object[] parameters)
         {
             int bestLength = -1;
+            int largestLengthWithMultipleAcceptedCtors = -1;
```

Comment:
I'm not sure I get what this variable is trying to track based on its name.

Maybe it would be more clear if it was `bool multipleBestLengthFound = false`.

And then set it to `true` / `false` as appropriate below.

Reply:
I didn't want it to be bool, because it only matters for us to throw ambiguous exception when for the best length (largest found length) we've found multiple ctors.